### DISCUSSION TITLE
Variants and optional attributes

### DISCUSSION THREAD

**Discussion by MoathCodes** [saleor/saleor#12071] [Q&A]
Hi,
So what I'm trying to do is:
Let's say I'm selling ice creams and I wanted to give the customer a choice whether he would like a strawberry on top of the ice creem or not `boolean`, if he wants the strawberry it will cost him 0.50$ more, there will be more options as well like the cup size, flavor, etc.

Looking at the storefront the way it was achieved is by creating a variant with attributes.

Does that mean that I have to create every single choice the customer might make?

For instance do I have to create a variant with large cup size with strawberry and another without, then repeate the process for each size?

This entire problem is probably because my lack of understanding for the attribute and variant system, I would appreciate an example or a document to refer to.
Thank you for your time.

**Comment by timuric** [comment]
@IronDragonDev it seems like you have many permutations of combinations, so making unique variants is probably not ideal. One way you could approach this is to use metadata and attributes. 

For example, you could create ice-cream product with attributes for possible extras. On front-end you could display attributes as options to users and when a customer adds an item to cart you can store configuration parameters as metadata on item line level. Once checkout is processed, you can display the recipe of order in a custom dashboard view.

In case additional options like strawberry cost extra, you could also store them as separate products and link them as options via attribute references to the parent product.

There are many ways to solve this problem using metadata, attributes, [custom prices](https://docs.saleor.io/docs/3.x/developer/checkout#setting-custom-prices-for-checkout-items)

Feel free to share more details about your use case.
